### PR TITLE
MODCLUSTER-754 Modify ResetRequestSourceImpl to send STOP request when proxy not aware of a context

### PR DESCRIPTION
**Pull request by TomasHofman**
https://issues.redhat.com/browse/MODCLUSTER-754
https://issues.redhat.com/browse/WFLY-16416

When an app server connects to a proxy for the first time or after a
disruption, it sends an INFO request to the proxy. Proxy responds with a
list of contexts that are registered for given server, and their states.
The context states returned from the proxy are compared to the actual
states of contexts on the app server, and if necessary, the server
should send ENABLE / STOP requests to bring the proxy to an up-to-date
state.

The change in this commit makes the app server send a STOP request for a
context in case when the proxy response to an INFO request doesn't
contain given context. The purpose of this STOP request is to enforce
context registration on the proxy side.

### CODE REVIEW

File: core/src/test/java/org/jboss/modcluster/mcmp/ResetRequestSourceTestCase.java [hunk 1]
```diff
@@ -49,7 +51,17 @@ public class ResetRequestSourceTestCase {
...
+
+    private final MCMPRequest configRequest = mock(MCMPRequest.class);
+    private final MCMPRequest enableRequest = mock(MCMPRequest.class);
+    private final MCMPRequest disableRequest = mock(MCMPRequest.class);
+    private final MCMPRequest stopRequest = mock(MCMPRequest.class);
+
+    private final static String JVM_ROUTE = "host1";
+    private final static String HOST_NAME = "host";
+    private final static String ALIAS1 = "alias1";
+    private final static String ALIAS2 = "alias2";
```
Comment (rhusar) [inline core/src/test/java/org/jboss/modcluster/mcmp/ResetRequestSourceTestCase.java:64]: Error:  /home/runner/work/mod_cluster/mod_cluster/core/src/test/java/org/jboss/modcluster/mcmp/ResetRequestSourceTestCase.java:61:19: 'static' modifier out of order with the JLS suggestions. [ModifierOrder]

Checkstyle hates the order here ;-)

**Review by rhusar — DISMISSED** [review]

File: core/src/test/java/org/jboss/modcluster/mcmp/ResetRequestSourceTestCase.java (hunk 1 above)
Comment (TomasHofman) [reply core/src/test/java/org/jboss/modcluster/mcmp/ResetRequestSourceTestCase.java:64]: Should be fixed now.

Comment (rhusar) [reply core/src/test/java/org/jboss/modcluster/mcmp/ResetRequestSourceTestCase.java:64]: @TomasHofman Nope, still isn't (the JLS proposes "private static final String" order.

Comment (TomasHofman) [reply core/src/test/java/org/jboss/modcluster/mcmp/ResetRequestSourceTestCase.java]: Ha, sorry, forgot `git add .` :)

Comment (TomasHofman) [reply core/src/test/java/org/jboss/modcluster/mcmp/ResetRequestSourceTestCase.java]: So, it should be fixed NOW :)

Comment (rhusar) [reply core/src/test/java/org/jboss/modcluster/mcmp/ResetRequestSourceTestCase.java]: Who needs `git add` when you have `git commit -a` ;-)

**Comment by TomasHofman** [comment]
Are there any further objections to this PR?

**Review by rhusar — APPROVED** [review]
LGTM

**Comment by rhusar** [comment]
Merged, thanks @TomasHofman !

**Comment by TomasHofman** [comment]
Thanks @rhusar ! I created backport for EAP 7.4.

**Pull request merged**